### PR TITLE
FIO. Refactor approve flow for pending request. Fix recordSend.

### PR DESCRIPTION
**Pull request by andreyvEze**

### CODE REVIEW

**Comment by paullinator** [comment]
Please fix conflicts

File: src/components/scenes/FioRequestListScene.js [hunk 1]
```diff
@@ -237,6 +260,61 @@ export class FioRequestList extends Component<StateProps, State> {
     )
   }
 
+  renderDropUp = (selectedFioPendingRequest: FioRequest) => {
+    const { onSelectWallet } = this.props
+    const { chain_code, token_code } = selectedFioPendingRequest.content
+    const allowedFullCurrencyCode = chain_code !== token_code && token_code && token_code !== '' ? [`${chain_code}:${token_code}`] : [chain_code]
+
```
Comment (andreyvEze) [inline src/components/scenes/FioRequestListScene.js:267]: add logic to set wallet if only one for the currency

File: src/components/scenes/FioRequestListScene.js [hunk 2]
```diff
@@ -237,6 +260,61 @@ export class FioRequestList extends Component<StateProps, State> {
...
+          const getFeeResult = await fioWalletByAddress.otherMethods.fioAction('getFee', {
+            endPoint: 'record_obt_data',
+            fioAddress: pendingRequest.payer_fio_address
+          })
+          if (getFeeResult.fee) {
+            showError(s.strings.fio_no_bundled_err_msg)
+            throw new Error(s.strings.fio_no_bundled_err_msg)
+          }
+        } catch (e) {
+          showError(s.strings.fio_get_fee_err_msg)
```
Comment (andreyvEze) [inline src/components/scenes/FioRequestListScene.js:303]: change to 'There was an issue getting fee. Please try again later.'

File: src/components/scenes/SendConfirmationScene.js [hunk 3]
```diff
@@ -217,8 +217,10 @@ export class SendConfirmation extends Component<Props, State> {
     const DESTINATION_TEXT = sprintf(s.strings.send_confirmation_to, destination)
     const ADDRESS_TEXT = sprintf(s.strings.send_confirmation_address, address)
     const fioAddress = this.props.guiMakeSpendInfo && this.props.guiMakeSpendInfo.fioAddress ? this.props.guiMakeSpendInfo.fioAddress : ''
-    const memo = this.props.guiMakeSpendInfo && this.props.guiMakeSpendInfo.memo ? this.props.guiMakeSpendInfo.memo : ''
-    const fio = this.props.guiMakeSpendInfo && this.props.guiMakeSpendInfo.isSendUsingFioAddress ? this.state.fioSender : null
+    const fio =
```
Comment (andreyvEze) [inline src/components/scenes/SendConfirmationScene.js:220]: rename to `fioSender`

File: src/components/scenes/SendConfirmationScene.js [hunk 4]
```diff
@@ -360,6 +348,7 @@ export class SendConfirmation extends Component<Props, State> {
                   selected={fio.fioAddress}
```
Comment (andreyvEze) [inline src/components/scenes/SendConfirmationScene.js:348]: remove `fio` checking, render `<SelectFioAddress />` and check inside of it

Comment (andreyvEze) [reply src/components/scenes/SendConfirmationScene.js]: done

File: src/components/scenes/SendConfirmationScene.js (hunk 3 above)
Comment (andreyvEze) [reply src/components/scenes/SendConfirmationScene.js]: renamed

File: src/components/scenes/FioRequestListScene.js (hunk 2 above)
Comment (andreyvEze) [reply src/components/scenes/FioRequestListScene.js:314]: changed

File: src/components/scenes/FioRequestListScene.js (hunk 1 above)
Comment (andreyvEze) [reply src/components/scenes/FioRequestListScene.js:279]: added

**Pull request merged**